Version 2

danielweck opened the pull request:
[Edit from Jiminy to list changes]

This new major version:

- removes responsive columns
- allows developers to apply the number of columns they see fit, w/o any restriction
- adds a flag to disable vertical-writing pagination for developers who prefer to implement their own
- removes a hack for older versions of Chromium, that created performance issues in some conditions
- replaces page margins user setting with line length user setting
- attempts to improve the handling of overflowing text
- re-implements the font-size user setting
- extends the application of the font-family user setting
- improves the default font-stacks
- updates recommandations for libre and open source fonts
- adds support for variable fonts
- adds a font-patch for Android dealing with rare occurrences of missing fonts in fixed-layout.

JayPanoz committed:
Remove -webkit-perspective hack (resolves #119)

JayPanoz committed:
Remove webkit-line-box-contain safeguard (resolves #113)

JayPanoz committed:
Remove responsive columns (resolves #143)

JayPanoz committed:
Init migration guide

JayPanoz committed:
Update migration guide

JayPanoz committed:
Implement line length user setting

Part of issue #130 resolution

JayPanoz committed:
Remove page margins setting

Resolves #130

JayPanoz committed:
Fix responsive columns removal

JayPanoz committed:
Update migration guide

JayPanoz committed:
Clarify pageGutter in migration

JayPanoz committed:
Add test for user line length

JayPanoz committed:
Remove page margins test

JayPanoz committed:
Update version

JayPanoz committed:
Remove postcss-custom-media from build script

JayPanoz committed:
Serve demo via express

So that it can be kept in sync more easily, and get around local files restrictions.

JayPanoz committed:
Replace font-size user setting with zoom

Resolves #112, resolves #95, closes #34

JayPanoz committed:
Correct docs for colCount and modules order

JayPanoz committed:
Update font-size and type-scale in docs

JayPanoz committed:
Add variable fonts pref submodules

JayPanoz committed:
Add regression tests

JayPanoz committed:
Update docs for variable fonts settings

JayPanoz committed:
Correct comment in fontWeight submodule

JayPanoz committed:
Correct font width test

JayPanoz committed:
Update docs following test corrections

JayPanoz committed:
Add bold in font-weight test

JayPanoz committed:
Change variable font for optical-sizing test

JayPanoz committed:
Modify font-width so that it can accept keywords

JayPanoz committed:
Adding selectors to break-word safeguard

JayPanoz committed:
Experiment with hidden and clip overflow

JayPanoz committed:
Add variable font tests in typeface tester util

JayPanoz committed:
Correct variations test and add utils to server

JayPanoz committed:
Add support for google variations in typetester

JayPanoz committed:
Update libre fonts doc with variations

JayPanoz committed:
Add header for optical sizing

Font-size depends on the font for the font-optical-sizing property so it may not be visible when only checking paragraphs

JayPanoz committed:
Add variations’ range for libre fonts

JayPanoz committed:
Add variable fonts to fonts review

JayPanoz committed:
Remove Source Sans Pro from var review

JayPanoz committed:
Add 2 recommanded fonts for a11y

JayPanoz committed:
Add variable fonts to md docs

JayPanoz committed:
Add MacOS variable fonts to review

JayPanoz committed:
Remove note about missing ™ in gFonts

JayPanoz committed:
Add Apple System fonts in var fonts

JayPanoz committed:
Add image alts in variable fonts

JayPanoz committed:
Update migration guide

JayPanoz committed:
Update migration guide

JayPanoz committed:
Update migration guide

JayPanoz committed:
Corrections in docs

JayPanoz committed:
Add var fonts to exploded EPUB

JayPanoz committed:
Add note about font collections in var docs

JayPanoz committed:
Improve default font stacks

Update of system fonts + linux

JayPanoz committed:
Document Windows 11 variable fonts

JayPanoz committed:
Add Windows 11 var fonts to default stacks

This intends to leverage optical sizing i.e. use heading and text adjustments depending on font size.

JayPanoz committed:
Update migration guide

JayPanoz committed:
Q&D improvement of type tester

Making it more interactive with inputs

JayPanoz committed:
Correct indent in fs normalize module

JayPanoz committed:
Correct namespaced selector for svg

Resolves #148

JayPanoz committed:
Update pagination model in docs

JayPanoz committed:
Update fragmented model for vertical-writing

JayPanoz committed:
Align vertical pagination with the default one

Remove --RS__pageGutter on :root for more consistency

JayPanoz committed:
Clarify pageGutter in docs

JayPanoz committed:
Add input in demo

So that you can add from the API w/o using debug tools

JayPanoz committed:
Allow colCount pref to accept any integer

JayPanoz committed:
Extend font-family setting overrides

Resolves #147

JayPanoz committed:
Create flag to disable vertical-writing pagination

As convenience to implementers’ preferring to implement theirs.

JayPanoz committed:
Update EBPAJ font patch in dist

JayPanoz committed:
Add script to update dist patches on build

JayPanoz committed:
Add Android Fonts Patch

Resolves #149

JayPanoz committed:
Update japanese font changes in docs

JayPanoz committed:
Update default font stacks in docs

JayPanoz committed:
Update EPUB doc

JayPanoz committed:
Update migration guide

Following engineering call agreed changes: number of columns, font-family override, flag to disable vertical writing pagination

JayPanoz committed:
Update migration guide with android fonts patch

JayPanoz committed:
Add clarification re. vertical writing pagination

Why implementers may want to disable the default and how it’s being done.

JayPanoz committed:
Update migration guide in EPUB docs

JayPanoz committed:
Merge branch 'develop' into v2

JayPanoz committed:
Rebuild dist and EPUB

mickael-menu commented on docs/CSS12-user_prefs.md:
```diff
@@ -306,22 +304,6 @@ Required flag: `--USER__advancedSettings: readium-advanced-on`
 
 Override class: User settings (should be applied by any means necessary)
 
-#### Type scale
```
Type scale is fully deprecated, there's nothing to replace it? I guess it comes from using `zoom`?

JayPanoz replied:
Well spotted, it should have been removed from the docs – it was only partly. 

So that was indeed tied to the font-size normalise we were using, as it was part of the font-size overrides we were applying to body copy elements (`h1`, `h2`, etc.).

Since `zoom` is global, we’re losing that ability.

mickael-menu replied:
Alright, I'm not sure it was used much anyway by integrators. We certainly didn't show it in the user preferences. Having a consistent font size is much more important to our users.

mickael-menu reviewed (commented):
Excellent job as usual! I left a few comments but saw nothing that would cause real trouble in the mobile toolkits.

JayPanoz commented on docs/CSS28-migration_guide.md:
```diff
@@ -0,0 +1,144 @@
+
+## Addition of a flag to disable vertical-writing pagination
+
+Given the limitations of columns fragmentation in vertical writing – progression direction from top to bottom instead of right to left, columns stacked on the `y-axis` –, some Reading Systems and apps have been disabling it, and implement their own logic.
```
You can think of it as a mere alias to scroll mode – it’s actually re-using the exact same declaration. It was just added for convenience, in case it makes more sense to other implementers to use this explicit flag instead of switching to scroll.

mickael-menu replied:
Alright, I think we will stick to the scroll mode on mobile as we have to show it in the user preferences anyway.

JayPanoz commented on docs/CSS28-migration_guide.md:
```diff
@@ -0,0 +1,144 @@
+
+The user setting should consequently be updated, either in the form of replacement or mimicking of the old one.
+
+It is indeed possible to re-implement the page margins user setting by using the new variable, given margin will adjust automatically to the line length, but this also means that your updated logic will have to work backwards.
```
Yeah that needs clarifying. Not sure what’s the best way to describe the process is but: 

- when using page margin you constrain line-length;
- when using line-length you constrain page margin.

So if you want to mimic a page-margin setting using line-length, you would reduce line-length to increase page margins, and increase line-length to reduce page margins. 

In v1, you would just decrease/increase the value of `--USER__pageMargin`.

mickael-menu replied:
Thanks, that was my understanding as well.

JayPanoz replied:
Will commit a clarification. Thanks for helping improve the wording here. 👍

JayPanoz commented on docs/CSS28-migration_guide.md:
```diff
@@ -0,0 +1,144 @@
+
+In version 2, ReadiumCSS provides support for three registered (read common) axes so that you can expose them as user settings. 
+
+1. `--USER__fontWeight`
```
It’s using `font-weight` so yeah it should work with all fonts, although with some you may be limited to 2 weight e.g. `400` and `700`. Others have more weights available so it would work almost as well as variable fonts, although with visible steps in the range.

JayPanoz committed:
Correct lineLength variable in CSS03-injection_and_pagination

Co-authored-by: Mickaël Menu <mickael.menu@gmail.com>

JayPanoz commented on docs/CSS03-injection_and_pagination.md:
````diff
@@ -136,19 +117,27 @@ You must account for this gap when scrolling.
 
 ```
---RS__maxLineLength
+--RS__defaultLineLength
````
The thing is I can’t give a thorough answer to this right now. I’m not sure it’s absolutely mandatory or not in ReadiumCSS as it was also originally used to force a recalc of multicol in Webkit after encountering weird issues while switching appearance/reading modes. 

In any case, I will have to test that when implementing Playground, so I preferred to keep it as the isolated testing I did isn’t necessarily representative of the real uses of ReadiumCSS.

JayPanoz committed:
Clarify migration + remove typeScale from docs

JayPanoz committed:
Remove font-size normalize from doc details

JayPanoz committed:
Simplify pagination

Remove 100vw as a condition for single-col from module and user-setting submodule. Using auto will prioritize column-count and resolve to 100% anyway.

JayPanoz committed:
Update dist

JayPanoz committed:
Run audit fix

JayPanoz committed:
Merge branch 'develop' into v2
